x-pack/plugins/transform/public/app/hooks/use_pivot_data.ts

```diff
@@ -235,7 +236,12 @@ export const usePivotData = (
   }, [indexPatternTitle, JSON.stringify([requestPayload, query, combinedRuntimeMappings])]);
 
   if (sortingColumns.length > 0) {
-    tableItems.sort(multiColumnSortFactory(sortingColumns));
+    const sortingColumnsWithTypes = sortingColumns.map((c) => ({
+      ...c,
+      // Since items might contain undefined/null values, we want to accurate find the data type
+      type: typeof tableItems.findIndex((item) => getNestedOrEscapedVal(item, c.id) !== undefined),
```

Comment:
I assume there is a mistake. You use `typeof` with the `.findIndex` method. `findIndex` always returns an index of an element in the array, which is a `number`.

Reply:
Thanks for catching that 🙏 Fixed here https://github.com/elastic/kibana/pull/124587/commits/b0a16fedb91f594219f8b932677cd9668031d62d